spotbugs/src/main/java/edu/umd/cs/findbugs/ba/obl/ObligationAnalysis.java

```diff
@@ -215,6 +215,17 @@ public void edgeTransfer(Edge edge, StateSet fact) throws DataflowAnalysisExcept
                             + edge.getSource().getLastInstruction());
                 }
                 fact.deleteObligation(comparedObligation, edge.getTarget().getLabel());
+
+                // closing a Statement closes the ResultSet
+                Obligation statement = database.getFactory().getObligationByName("java.sql.Statement");
+                if (comparedObligation.equals(statement)) {
+                    Obligation resultSet = database.getFactory().getObligationByName("java.sql.ResultSet");
+                    fact.deleteObligation(resultSet, edge.getTarget().getLabel());
+                    if (DEBUG_NULL_CHECK) {
+                        System.out.println("Deleting " + resultSet.toString() + " on edge from comparison "
```

Comment:
Can this be a logger statement instead of STDOUT?

Reply:
Thank you very much for your valuable comments and suggestions.
I fixed all issues in the latest commit :)